source/Ship.cpp

```diff
@@ -630,7 +618,15 @@ void Ship::FinishLoading(bool isNewInstance)
 				count -= eit->second;
 
 			if(count)
-				armament.Add(it.first, count);
+				count -= armament.Add(it.first, count);
+			if(count)
+			{
+				string warning = modelName;
+				if(!name.empty())
+					warning += " \"" + name + "\"";
```

Comment:
At this point we have a number of "log an error for this ship with its name" blocks; let's consolidate the logic for determining the name into a function in the anonymous namespace.

Reply:
Good idea, I added a helper function for this.